core/controllers/domain_objects_validator.py

```diff
@@ -178,3 +178,13 @@ def validate_aggregated_stats(aggregated_stats):
                 raise base.BaseHandler.InvalidInputException(
                     '%s not in state stats mapping of %s in aggregated '
                     'stats dict.' % (state_stats_property, state_name))
+
+
+def validate_params_dict(params):
+    """validates params dict
+
+    Args:
+        params: dict. Data that needs to be validated.
+    """
+    for value in params.values():
```

Comment:
1. @SD-13 Here you should verify the type of key to be string instead of value -- based on [this comment](https://github.com/oppia/oppia/blob/8b1986ef0746ebec1d52c3243c078cc58bdd15a0/core/templates/pages/exploration-player-page/services/stats-reporting.service.ts#L352) it is a statename.

2. Also just for confirmation, if keys are state name then its values can be a State Object in dict format. If this is the case can we use validate method for state object?

Reply:
Please take a look [here](https://github.com/oppia/oppia/pull/14208/files#r757851606).